frontend/scripts/check-patternfly-modules.ts

```diff
@@ -0,0 +1,119 @@
+/* eslint-disable no-console */
+import * as lockfile from '@yarnpkg/lockfile';
+import { readFileSync } from 'fs';
+import { basename } from 'path';
+import chalk from 'chalk';
+import * as semver from 'semver';
+
+console.log(`Checking ${chalk.yellow('yarn.lock')} for PatternFly module resolutions...`);
+
+const lockFile = lockfile.parse(readFileSync('yarn.lock', 'utf8'));
+
+if (lockFile.type !== 'success') {
+  console.error(`Failed to parse yarn.lock file: type is ${lockFile.type}`);
+  process.exit(1);
+}
+
+/** List of packages to check along with their expected semver ranges */
+const PKGS_TO_CHECK: Array<{ name: string; semver: string }> = [
+  { name: '@patternfly-5/patternfly', semver: '*' },
+  { name: '@patternfly/patternfly', semver: '*' },
+  { name: '@patternfly/quickstarts', semver: '*' },
+  { name: '@patternfly/react-catalog-view-extension', semver: '*' },
+  { name: '@patternfly/react-charts', semver: '*' },
+  { name: '@patternfly/react-code-editor', semver: '*' },
+  { name: '@patternfly/react-component-groups', semver: '*' },
+  { name: '@patternfly/react-core', semver: '*' },
+  { name: '@patternfly/react-data-view', semver: '*' },
+  { name: '@patternfly/react-icons', semver: '*' },
+  { name: '@patternfly/react-log-viewer', semver: '*' },
+  { name: '@patternfly/react-styles', semver: '*' },
+  { name: '@patternfly/react-table', semver: '*' },
+  { name: '@patternfly/react-templates', semver: '*' },
+  { name: '@patternfly/react-tokens', semver: '*' },
+  { name: '@patternfly/react-topology', semver: '*' },
+  { name: '@patternfly/react-user-feedback', semver: '*' },
+  { name: '@patternfly/react-virtualized-extension', semver: '*' },
+];
+
+const SCOPES_TO_CHECK = new Set(PKGS_TO_CHECK.map((pkg) => pkg.name.split('/')[0]));
+
+/** Get the package name and version from a package key in yarn.lock */
+const parsePackageName = (pkgKey: string): { pkgName: string } => {
```

Comment:
Nit - we could just return a `string` instead of `{ pkgName: string }` object.

Also, can you give an example on `pkgKey` difference vs `pkgName` ?

Reply:
`pkgkey` is `@patternfly/patternfly@^6.2.3`, `pkgname` is `@patternfly/patternfly`. previously this also parsed the version from pgkey, but it wasn't used so I removed that part